### ISSUE TITLE
MongoDB

### ISSUE THREAD

**Issue by AlexDaniel** [perl6/ecosystem-unbitrot#475]
Module [MongoDB](https://modules.perl6.org/dist/MongoDB) cannot be installed (`ZefFailure`), perhaps it has some failing tests.

* Tickets are opened/closed in this repo automatically (though not immediately).
* If you can **install the module without any problems**, add `works for me` label, leave a comment saying that it works for you and mention any details that you feel are important.
* If it **needs a native library**, put `native dependency` label, describe what you did to install it and ensure that same instructions are present in the README file of the module (otherwise submit a pull request). Also try to update [this wiki page](https://github.com/perl6/ecosystem-unbitrot/wiki).
* If the **module is broken**, try to fix it and send a PR. Add `PR sent` label.
* If there is a **problem in one of the dependencies**, add `failing dependency` label and write a comment explaining the situation. Feel free to work on the corresponding ticket for the failing dependency.
* It is a good idea to **assign yourself** to this ticket if you're working on it (to make sure two or more people are not working on the same ticket at the same time).
* **Once you are done**, [search for a next ticket](https://github.com/perl6/ecosystem-unbitrot/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22PR+sent%22+-label%3A%22deprecated+module%22+-label%3A%22native+dependency%22).

If you can't self-assign yourself or attach a label, please let us know on [#perl6 channel on freenode](https://perl6.org/irc) or just leave a comment here. We will try to give you privileges as fast as possible.

Output:

````````````````````````````````````````````````````````
===> Searching for: MongoDB
«timed out after 600 seconds»
````````````````````````````````````````````````````````

Ping @MARTIMM

Previous ticket: #130

**Comment by AlexDaniel** [comment]
I think the tests are just really slow?

**Comment by Altai-man** [comment]
No, I don't think so. Hang after `Searching for` is a zef hang, no? The module wasn't even downloaded to test it.

**Comment by jonathanstowe** [comment]
@JJ noticed that the modules that this was happening to have a ```git:///``` source-url in their META.  Is it perhaps something to do with the network this is being tested on?

**Comment by JJ** [comment]
Those git URLs work well if you use, well, git. They don't work if you try to download using any web client.  It's better to change them to an HTTP URI, since that works in both cases.

**Comment by MARTIMM** [comment]
first I want to apologize for the late reaction on this, I totally missed the action...
Following the suggestion of @jonathanstowe, I changed the source URL and it downloads on my Linux system using zef. The tests are failing badly but I now see why it fails.

**Comment by MARTIMM** [comment]
The latest version of MongoDB does install and test on my system. Needed to change a lot and tried something new (for me), a Build.pm6 is made to download a server exec and a sandbox is created. Still, I need to do things for windows though. The installation log is shown below
```
> zef -v install MongoDB
===> Searching for: MongoDB
===> Updating cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
===> Updating p6c mirror: http://ecosystem-api.p6c.org/projects1.json
===> Updated cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
===> Updated p6c mirror: http://ecosystem-api.p6c.org/projects1.json
===> Found: MongoDB:ver<0.43.4.3>:api<0> [via Zef::Repository::Ecosystems<p6c>]
===> Fetching [OK]: MongoDB:ver<0.43.4.3>:api<0> to /home/marcel/.zef/tmp/1553201939.1691.3574/mongo-perl6-driver.git
===> Extraction [OK]: MongoDB to /home/marcel/.zef/store/mongo-perl6-driver.git
===> Building: MongoDB:ver<0.43.4.3>:api<0>
Command: /opt/Perl6/rakudo/install/bin/perl6 -Ilib -e require '/home/marcel/.zef/store/mongo-perl6-driver.git/1a1a5ce0685dc4fdfe25db1330a2ddfc3d3bc87a/Build.pm6'; ::('Build').new.build('/home/marcel/.zef/store/mongo-perl6-driver.git/1a1a5ce0685dc4fdfe25db1330a2ddfc3d3bc87a') ?? exit(0) !! exit(1);
mongodb-linux-x86_64-3.6.9/bin/mongod
===> Building [OK] for MongoDB:ver<0.43.4.3>:api<0>
===> Testing: MongoDB:ver<0.43.4.3>:api<0>
t/099-start-servers.t ... ok
    # BSON::Document.new((
    #   ok => 0e0,
    #   errmsg => "a collection 'test.cl1' already exists",
    #   code => 48,
    #   codeName => "NamespaceExists",
    # ))
    # 
t/200-Database.t ........ ok
t/300-Collection.t ...... ok
t/301-Collection.t ...... ok
t/400-run-command.t ..... ok
t/401-rc-query-write.t .. ok
t/450-find.t ............ ok
t/998-stop-servers.t .... ok
All tests successful.
Files=8, Tests=28,  78 wallclock secs
Result: PASS
===> Testing [OK] for MongoDB:ver<0.43.4.3>:api<0>
===> Installing: MongoDB:ver<0.43.4.3>:api<0>
===> Install [OK] for MongoDB:ver<0.43.4.3>:api<0>

4 bin/ scripts [mongodb-accounting.pl6 start-servers.pl6 make-replicaset.pl6 stop-servers.pl6] installed to:
/mnt/Data/opt/Perl6/rakudo/install/share/perl6/site/bin
```
